drivers/photo-electronic/adpd410x/adpd410x.h

```diff
@@ -0,0 +1,1244 @@
+
+/**
+ * @struct adpd410x_init_param
+ * @brief Device driver initialization structure
+ */
+struct adpd410x_init_param {
+	/** Physical communication */
+	union phy_comm_init_param dev_ops_init;
+	/** Device option */
+	enum adpd410x_supported_dev dev_type;
+	/** Device clock option */
+	enum adpd410x_clk_opt clk_opt;
+	/** GPIO 0 initialization */
+	struct gpio_init_param gpio0;
+	/** GPIO 1 initialization */
+	struct gpio_init_param gpio1;
+	/** GPIO 2 initialization */
+	struct gpio_init_param gpio2;
+	/** GPIO 3 initialization */
+	struct gpio_init_param gpio3;
+	/** External low frequency oscillator frequency, if applicable */
+	uint32_t ext_lfo_freq;
+};
+
+/**
+ * @struct adpd410x_dev
+ * @brief Device driver handler
+ */
+struct adpd410x_dev {
+	/** Physical communication */
+	union phy_comm_dev dev_ops;
+	/** Device option */
+	enum adpd410x_supported_dev dev_type;
+	/** Device clock option */
+	enum adpd410x_clk_opt clk_opt;
+	/** GPIO 0 handler */
+	struct gpio_desc *gpio0;
+	/** GPIO 1 handler */
+	struct gpio_desc *gpio1;
+	/** GPIO 2 handler */
+	struct gpio_desc *gpio2;
+	/** GPIO 3 handler */
+	struct gpio_desc *gpio3;
+	/** External low frequency oscillator frequency, if applicable */
+	uint32_t ext_lfo_freq;
+};
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+/** Read device register. */
+int32_t adpd410x_reg_read(struct adpd410x_dev *dev, uint16_t address,
+			  uint16_t *data);
+
+/** Write device register. */
+int32_t adpd410x_reg_write(struct adpd410x_dev *dev, uint16_t address,
+			   uint16_t data);
+
+/** Do a read and write of a register to update only part of a register. */
+int32_t adpd410x_reg_write_mask(struct adpd410x_dev *dev, uint16_t address,
+				uint16_t data, uint16_t mask);
+
+/** Do a software reset. */
+int32_t adpd410x_reset(struct adpd410x_dev *dev);
+
+/** Set operation mode. */
+int32_t adpd410x_set_opmode(struct adpd410x_dev *dev,
+			    enum adpd410x_opmode mode);
+
+/** Set number of active time slots. */
+int32_t adpd410x_set_last_timeslot(struct adpd410x_dev *dev,
+				   uint8_t timeslot_no);
+
+/** Set device sampling frequency. */
+int32_t adpd410x_set_sampling_freq(struct adpd410x_dev *dev,
+				   uint32_t sampling_freq);
+
+/** Setup an active time slot. */
+int32_t adpd410x_timeslot_setup(struct adpd410x_dev *dev,
+				enum adpd410x_timeslots timeslot_no,
+				struct adpd410x_timeslot_init *init);
+
+/** Get number of bytes in the device FIFO. */
+int32_t adpd410x_get_fifo_bytecount(struct adpd410x_dev *dev, uint16_t *bytes);
+
+/** Get a full data packet from the device containing data from all active time
+ *  slots. */
+int32_t adpd410x_get_data(struct adpd410x_dev *dev, uint32_t *data);
+
+/** Setup the device and the driver. */
+int32_t adpd410x_setup(struct adpd410x_dev **device,
```

Comment:
Rename it to `adpd410x_init` for consistency with other drivers.

Reply:
I see that sometimes `_setup` is used.

Reply:
I think we use _setup because of the drivers on linux

Reply:
So is it OK to remain to setup? Or should we change to init?

Reply:
Is ok `setup`